agent/http.go

```diff
@@ -551,6 +616,11 @@ func (s *HTTPServer) parseConsistency(resp http.ResponseWriter, req *http.Reques
 		fmt.Fprint(resp, "Cannot specify ?stale with ?consistent, conflicting semantics.")
 		return true
 	}
+	if b.UseCache && b.RequireConsistent {
+		resp.WriteHeader(http.StatusBadRequest)
```

Comment:
Should this function instead return an error and be refactored to return the BadRequestError?

Maybe not as this change would need to be propagated through a couple other places. Just a thought

Reply:
Could but I followed the precedent of all other helpers in this file - I think if we want to clean that up we should do it all at once separately.

Reply:
Yeah, thats probably best.